Content.Shared/TapeRecorder/SharedTapeRecorderSystem.cs

```diff
@@ -0,0 +1,610 @@
+    {
+        //Is the tape damaged?
+        if (HasComp<FitsInTapeRecorderComponent>(tapeCassette))
+            return;
+
+        //Are we using a pen or screwdriver?
+        if (!_tagSystem.HasAnyTag(args.Used, "Screwdriver", "Write"))
+            return;
+
+        _doAfterSystem.TryStartDoAfter(new DoAfterArgs(EntityManager, args.User, tapeCassette.Comp.RepairDelay, new TapeCassetteRepairDoAfterEvent(), tapeCassette, target: args.Target, used: args.Used)
+        {
+            BreakOnMove = true,
+            NeedHand = true
+        });
+    }
+
+    /// <summary>
+    /// Repair a damaged tape
+    /// </summary>
+    protected void OnTapeCassetteRepair(Entity<TapeCassetteComponent> tapeCassette, ref TapeCassetteRepairDoAfterEvent args)
+    {
+        if (args.Handled || args.Cancelled || args.Args.Target == null)
+            return;
+
+        //Cant repair if not damaged
+        if (HasComp<FitsInTapeRecorderComponent>(tapeCassette))
+            return;
+
+        _appearanceSystem.SetData(tapeCassette, ToggleVisuals.Toggled, false);
+        AddComp<FitsInTapeRecorderComponent>(tapeCassette);
+        args.Handled = true;
+    }
+
+    /// <summary>
+    /// When the cassette has been damaged, corrupt and entry and unspool it
+    /// </summary>
+    protected void OnDamagedChanged(Entity<TapeCassetteComponent> tapeCassette, ref DamageChangedEvent args)
+    {
+        if (args.DamageDelta == null || args.DamageDelta.GetTotal() < 5)
+            return;
+
+        _appearanceSystem.SetData(tapeCassette, ToggleVisuals.Toggled, true);
+
+        RemComp<FitsInTapeRecorderComponent>(tapeCassette);
+        CorruptRandomEntry(tapeCassette);
+    }
+
+    protected void OnTapeExamined(Entity<TapeCassetteComponent> tapeCassette, ref ExaminedEvent args)
+    {
+        if (args.IsInDetailsRange)
+        {
+            if (HasComp<FitsInTapeRecorderComponent>(tapeCassette))
+            {
+                var positionPercentage = Math.Floor(tapeCassette.Comp.CurrentPosition / tapeCassette.Comp.MaxCapacity.TotalSeconds * 100);
+                var tapePosMsg = Loc.GetString(tapeCassette.Comp.TextExamine,
+                    ("position", positionPercentage)
+                    );
+                args.PushMarkup(tapePosMsg);
+            }
+           else
+            {
+                args.PushMarkup(Loc.GetString(tapeCassette.Comp.TextDamaged));
+            }
+        }
+    }
+
+    protected void OnRecorderExamined(Entity<TapeRecorderComponent> tapeRecorder, ref ExaminedEvent args)
+    {
+        if (args.IsInDetailsRange)
+        {
+            switch (tapeRecorder.Comp.Mode)
+            {
+                case TapeRecorderMode.Playing:
+                    args.PushMarkup(Loc.GetString(tapeRecorder.Comp.TextModePlaying));
+                    break;
+                case TapeRecorderMode.Stopped:
+                    args.PushMarkup(Loc.GetString(tapeRecorder.Comp.TextModeStopped));
```

Comment:
Just do
```c#
args.PushMarkup(Loc.GetString(tapeRecorder.Comp.Mode switch {
    TapeRecorderMode.Playing => tapeRecorder.Comp.TextModePlaying,
    ...
}));
```
instead of duplicating `args.PushMarkup(Loc.GetString())` 4 times.